File: kotlin/isogram/src/main/kotlin/Isogram.kt
object Isogram {
    fun isIsogram(input: String): Boolean {
        var lowerCase = input.toLowerCase()

        for(char in lowerCase) {
            if (lowerCase.count { it == char && it in 'a'..'z' && lowerCase.indexOf(it) > -1} > 1) return false
        }

        return true
    }
}

File: csharp/isogram/Isogram.cs
using System;
using System.Linq;

public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        var distinctCount = word
            .ToLower()
            .Where(@char => char.IsLetter(@char))
            .Distinct().Count();

        var originalCount = word.
            Where(@char => char.IsLetter(@char))
            .Count();

        return distinctCount == originalCount;
    }
}
